modules/core/src/main/java/org/locationtech/jts/index/strtree/STRtree.java

```diff
@@ -373,5 +402,91 @@ public Object nearestNeighbour(Envelope env, Object item, ItemDistance itemDist)
           ((ItemBoundable) minPair.getBoundable(1)).getItem()
       };
   }
+  private Object[] nearestNeighbour(BoundablePair initBndPair, double maxDistance, int k) 
```

Comment:
@dr-jts Yes, you are right. If k = 1, the KNN method has the same result and same performance with the original NN(1) method because the maintained KNN Queue just becomes 1NN. However, their APIs are a little bit different. The return of the original NN(1) method is a different array in the following format:[Nearest item, Query Point]. The new NN(K) method returns an array [Nearest item1, item2, item3,...., item k] without including Query Point.